### PR TITLE
Expose time spent in SplitSource.getNextBatch as TimeStat in JMX

### PR DESCRIPTION
**Pull request by findepi**
Fixes presto's #6710

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java [hunk 1]
```diff
@@ -318,7 +322,9 @@ public StageInfo getStageInfo(Supplier<Iterable<TaskInfo>> taskInfosSupplier, Su
 
     public void recordGetSplitTime(long startNanos)
     {
+        long elapsedNanos = System.nanoTime() - startNanos;
         getSplitDistribution.add(System.nanoTime() - startNanos);
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java]: use elapsedNanos

File: presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java [hunk 2]
```diff
@@ -75,12 +78,13 @@
     private final AtomicLong peakMemory = new AtomicLong();
     private final AtomicLong currentMemory = new AtomicLong();
 
-    public StageStateMachine(StageId stageId, URI location, Session session, PlanFragment fragment, ExecutorService executor)
+    public StageStateMachine(StageId stageId, URI location, Session session, PlanFragment fragment, ExecutorService executor, SplitSchedulerStats schedulerStats)
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java:81]: split into multiple lines?

File: presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java (hunk 1 above)
Comment (fiedukow) [inline presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java]: this variable is already extracted :D

File: presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java (hunk 2 above)
Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java:81]: what's the line length limit?

**Pull request closed without merging**